app/models/services/commitment_props.rb

```diff
@@ -0,0 +1,322 @@
+                  'Certification of products',
+                  'Recreation',
+                  'Academic research'
+                ]
+              },
+              {
+                type: 'checkbox',
+                name: 'managerIds',
+                title:
+                  'Who will be responsible for deciding how the area is being managed?',
+                description: 'Multi select option',
+                # defaultValue: [],
+                choices: [
+                  'Indigenous peoples',
+                  'Local communities',
+                  'For-profit organisations',
+                  'Non-profit organisations',
+                  'Individual landowners',
+                  'Collaborative governance',
+                  'Joint governance',
+                  'Sub-national ministry or agency',
+                  'Other'
+                ],
+                otherText: 'Other'
+              },
+              {
+                type: 'comment',
+                name: 'jointGovernanceType',
+                visibleIf: "{managerIds} contains 'Joint governance'",
+                title: 'Specify the type of joint governance',
+                hideNumber: true
+              },
+              {
+                type: 'text',
+                name: 'responsibleGroup',
+                title: 'Name of the group responsible for decisions'
+              }
+            ],
+            title: 'General Info',
+            description:
+              '‘Commitment’ refers to what you intend to do, for example establishing a new protected area, or working to improve the management of an existing area. ‘Area’ refers to the geographical area that is (or will be) under management as a result of this commitment.’'
+          },
+          {
+            name: 'locationPage',
+            elements: [
+              {
+                type: 'tagbox',
+                name: 'countryIds',
+                title: 'Select commitment location by country',
+                description: 'Multi select option',
+                # defaultValue: [],
+                choicesByUrl: {
+                  url: 'https://surveyjs.io/api/CountriesExample'
+                },
+                optionsCaption: '-Select a country-'
+              },
+              {
+                type: 'expression',
+                name: 'latlongHeading',
+                title: 'Please provide the latitude and longitude of the area'
+              },
+              {
+                type: 'text',
+                name: 'latitude',
+                title: 'Latitude',
+                titleLocation: 'left',
+                hideNumber: true
+              },
+              {
+                type: 'text',
+                name: 'longitude',
+                title: 'Longitude',
+                titleLocation: 'left',
+                hideNumber: true
+              },
+              {
+                type: 'file',
+                name: 'geospatialFilename',
```

Comment:
this is spatialData currently. 'filename' is what is in the erd, but is wrong. probably the name should be geospatialFile here, and you can just update ```has_one_attached :spatial_data``` in Commitment to ```has_one_attached :geospatial_file```